### PR TITLE
Adding JRF domain into ItJrfDomainInPV 

### PR DESCRIPTION
**Pull request by maggiehe00**
This PR is to 1) add JRF domain on PV into test class ItJrfDomainInPV and verify the servers startup and services are created; 2) fix intermittent DB pod creation failure on the remote cluster

Jenkins cluster test result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-model-in-image-tests-1/43/testReport/
Kind cluster test result:
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/506/testReport/oracle.weblogic.kubernetes/

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 1]
```diff
@@ -41,7 +79,10 @@
...
   private static final String ORACLEDBURLPREFIX = "oracledb.";
   private static final String ORACLEDBSUFFIX = ".svc.cluster.local:1521/devpdb.k8s";
-
+  private static final String RCUSYSUSERNAME = "sys";
+  private static final String RCUSYSPASSWORD = "Oradoc_db1";
+  private static final String RCUSCHEMAUSERNAME = "myrcuuser";
+  private static final String RCUSCHEMAPASSWORD = "Oradoc_db1";
 
   private static String dbUrl = null;
   private static int dbPort = getNextFreePort(30000, 32767);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:88]: Do not evaluate and assign this until you are ready to use it.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 2]
```diff
@@ -65,38 +110,31 @@
    * @param namespaces injected by JUnit
    */
   @BeforeAll
-  public static void initAll(@Namespaces(1) List<String> namespaces) {
+  public static void initAll(@Namespaces(3) List<String> namespaces) {
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:113]: Now you need to initialize the logger in this method and remove the implements LoggedTest.

See https://github.com/oracle/weblogic-kubernetes-operator/pull/1765

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java [hunk 3]
```diff
@@ -114,21 +152,196 @@ public static void initAll(@Namespaces(1) List<String> namespaces) {
...
+      logger.info("Waiting for managed server pod {0} to be ready in namespace {1}",
+          managedServerPodNamePrefix + i, jrfDomainNamespace);
+      checkPodReady(managedServerPodNamePrefix + i, domainUid, jrfDomainNamespace);
+    }
+
+    // verify managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Checking managed server service {0} is created in namespace {1}",
+          managedServerPodNamePrefix + i, jrfDomainNamespace);
+      checkServiceExists(managedServerPodNamePrefix + i, jrfDomainNamespace);
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:299]: Move the checkServiceExists before checkPodReady for both admin and managed server pods.

There are instances where when service creation fails, the pods never become ready and the error log messages will be misleading

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 4]
```diff
@@ -1080,6 +1105,31 @@ public static void createSecretWithUsernamePassword(String secretName,
...
+  /**
+   * Create a RCU secret with username, password and sys_username, sys_password in the specified namespace.
+   *
+   * @param secretName secret name to create
+   * @param namespace namespace in which the secret will be created
+   * @param username username in the secret
+   * @param password passowrd in the secret
+   */
+  public static void createRcuSecretWithUsernamePassword(String secretName, String namespace,
+      String username, String password, String sysUsername, String sysPassword) {
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:1117]: missing params sysUsername and sysPassword in javadoc

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 5]
```diff
@@ -1660,8 +1710,223 @@ public static Path generateFileFromTemplate(
...
   }
+
+  /**
+  * Create a persistent volume.
+   *
+   * @param pvName name of the persistent volume to create
+   * @param domainUid domain UID
+   * @param className name of the class to call this method
+   */
+  public static void createPV(String pvName, String domainUid, String className) {
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:1725]: You need to get logger in each method it looks like.

LoggingFacade logger = getLogger();

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java [hunk 6]
```diff
@@ -1660,8 +1710,223 @@ public static Path generateFileFromTemplate(
...
+  /**
+   * Create a persistent volume claim.
+   *
+   * @param pvName name of the persistent volume
+   * @param pvcName name of the persistent volume claim to create
+   * @param domainUid UID of the WebLogic domain
+   * @param namespace name of the namespace in which to create the persistent volume claim
+   */
+  public static void createPVC(String pvName, String pvcName, String domainUid, String namespace) {
+    logger.info("creating persistent volume claim for pvName {0}, pvcName {1}, "
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:1768]: create logger, same applies for methods you are adding.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 2 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:113]: Yeah, when I merged to the latest develop branch I can see the conflicts.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 1 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java]: Fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 2 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java:113]: Fixed.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java (hunk 3 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfDomainInPV.java]: Fixed.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 4 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:1139]: Fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 5 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:1791]: Done

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java (hunk 6 above)
Comment (maggiehe00) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonTestUtils.java:1838]: Done.

**Pull request merged**